plugins/modules/sns_topic.py

```diff
@@ -365,9 +365,11 @@ def __init__(self,
     def _create_topic(self):
         attributes = {'FifoTopic': 'false'}
         tags = []
+        endpoint = self.module.region
 
         if self.topic_type == 'fifo':
-            attributes['FifoTopic'] = 'true'
+            if "us-gov" not in endpoint:
```

Comment:
Using `us-gov` since this is unlikely that AWS would releases non-US GovCloud regions -- at least based on my reading. I could be wrong though.

Reply:
I think this check needs to go up above line 366.

The change as implemented here is going to result in
1) The attribute still being set (line 366), but always being passed as "False" in gov-cloud (even if someone tries to create a FIFO topic).  This is the behaviour that's already causing problems.
2) In the event that gov-cloud does bring in support, the module will try to create a Topic, named as <topic>.fifo, however, without the `FifoTopic=True` attribute being passed.

I also believe we shouldn't hard code a region related "fail_json" if someone does try to create a FIFO Topic in Gov-Cloud:  Doing so means that we need to modify and release a new version of the code when Amazon enables the feature, and Amazon's constantly adding services and features.  While there may be some fundamental security reason why FIFO support is disabled, it could simply be pending some form of audit/approvals process.

Reply:
@tremble if we shouldn't hard code the region, how should be proceed about disabling FIFO topic when the user is using GovCloud?
There's nothing in the SNS API that exposes its capabilities, we only have the documentation that mentions that FIFO topics are not supported for GovCloud regions, but there's isn't much information out there.

Reply:
Amazon's generally pretty good about the APIs being backwards compatible, so rarely add new mandatory attributes/parameters once the API's been published.

Looking at the docs, the FifoTopic attribute is **optional** in all regions, and I presume will be assumed to be False if not set.  As such, it should only be set at all when trying to create a FIFO topic.  So do similar things where very new versions of botocore are needed for a feature.

I'd recommend leaving a comment in the code along the lines of

```
# Never set FifoTopic=False some regions (including GovCloud) don't support the atttibute being set, even to False.
```

Don't try and sort of hack something so you get an SNS topic, but not a real FIFO topic, when someone tries to create a FIFO topic in an unsupported region.  Update the documentation and just let the API fail if the region doesn't support the feature.